IOS/SwiftSeedProject/SwiftSeedProject/CoreDataManager.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  CoreDataManager.swift
+//  SwiftSeedProject
+//
+//  Created by Lucas Pelizza on 3/30/17.
+//  Copyright © 2017 Making Sense. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+class CoreDataManager {
+    
+    typealias CoreDataManagerCompletion = () -> ()
+    private let modelName: String
+    fileprivate let completion: CoreDataManagerCompletion
+    
+    private var persistentStoreURL: NSURL {
+
+        let storeName = "\(modelName).sqlite"
+        let fileManager = FileManager.default
+        
+        let documentsDirectoryURL = fileManager.urls(for: .documentDirectory, in: .userDomainMask)[0]
+        
+        return documentsDirectoryURL.appendingPathComponent(storeName) as NSURL
+    }
+    
+    private lazy var managedObjectModel: NSManagedObjectModel? = {
+
+        guard let modelURL = Bundle.main.url(forResource: self.modelName, withExtension: "momd") else {
+            return nil
+        }
+        
+        let managedObjectModel = NSManagedObjectModel(contentsOf: modelURL)
+        
+        return managedObjectModel
+    }()
+    
+    private lazy var persistentStoreCoordinator: NSPersistentStoreCoordinator? = {
+        guard let managedObjectModel = self.managedObjectModel else {
+            return nil
+        }
+        
+        let persistentStoreCoordinator = NSPersistentStoreCoordinator(managedObjectModel: managedObjectModel)
```

Comment:
Are you planning on adding code in-between these two lines? Otherwise, I'd just return the object instead of assigning it and then returning it.